sandswitches/manage.py

```diff
@@ -63,29 +82,99 @@ def __init__(self, rfile, etree, sftp, fscli, log):
         self.root = etree.getroot()
         self.file = rfile
         self.sftp = sftp
-        self.fscli = fscli
+        self.cli = fscli
         self.log = log
         self._touched = []
 
+    @property
+    def fscli(self):
+        """Alias for self.cli
+        """
+        return self.cli
+
     def revert(self):
         """Revert all changes to the root config file.
         """
         self.log.debug("restoring '{}'".format(self.file.path))
         self.file.restore()
 
     def commit(self):
+        """Commit the working XML etree to the remote FreeSWITCH config.
+        """
         now = time.time()
         self.log.info("saving '{}'".format(self.file.path))
         # write local copy
         self.sftp.putfo(
             BytesIO(etree.tostring(self.etree, pretty_print=True)),
             self.file.path
         )
-        # self.file.write(etree.tostring(self.etree, pretty_print=True))
         self.fscli('reloadxml')
         log.debug("freeswitch.xml commit took {} seconds"
                   .format(time.time() - now))
 
+    def sofia_status(self):
+        """Return status data from sofia in a nicely organized dict.
+        """
+        # remove '===' "lines"
+        lines = [
+            line for line in self.fscli('sofia', 'status').splitlines()
+            if '===' not in line]
+        # pop summary line
+        lines.pop(-1)
+
+        # build component to status maps
+        profiles = {}
+        gateways = {}
+        aliases = {}
+
+        colnames = [name.lower() for name in lines.pop(0).split()]
+        iname = colnames.index('name')
+        colnames.remove('name')
+        for line in lines:
+            fields = [field.strip() for field in line.split('\t')]
+            name = fields.pop(iname)
+            row = {k.lower(): v for k, v in zip(colnames, fields)}
+            tp = row.pop('type')
+            if tp == 'profile':
+                profiles[name] = row
+            elif tp == 'gateway':
+                profname, gwname = name.split("::")
+                row['profile'] = profname
+                gateways[gwname] = row
+            elif tp == 'alias':
+                aliases[name] = row
+
+        return {'profiles': profiles, 'gateways': gateways, 'aliases': aliases}
+
+    def get_users(self, **kwargs):
+        """Return all directory users in a map keyed by domain name.
+        """
+        args = []
+        allowed = ('domain', 'group', 'user', 'context')
+        if kwargs:
+            for argname, value in kwargs.items():
+                if argname not in allowed:
+                    raise ValueError(
+                        '{} is not a valid argument to list_users'.format(
+                            argname)
+                    )
+                args.append('{} {}'.format(argname, value))
+
```

Comment:
I would remove the if kwargs condition, it's redundant, try to iterate directly .. if non extra arguments are passed then kwargs.items() is equals to []